packages/create-react-app-demo/src/docs-chat/index.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { AIRoot, ChatMessage, conversationAtom, modelCallInProgress } from './ai.tsx';
+import { useAtom } from 'jotai';
+import ResultContainer from '../ResultContainer.tsx';
+
+function ConversationItem({ response }: { response: ChatMessage }) {
+  const emoji = response.type === 'user' ? '👤' : '🤖';
+  return (
+    <div>
+      {emoji}: {response.content}
+    </div>
+  );
+}
+
+function ConversationHistory() {
+  const [conversation, setConversation] = useAtom(conversationAtom);
+  const [callInProgress] = useAtom(modelCallInProgress);
+
+  function handleInputSubmit(event: React.FormEvent<HTMLFormElement>) {
+    // @ts-expect-error
+    const element = event.target.elements.message;
+    event.preventDefault();
+    setConversation((prev) => [
+      ...prev,
+      {
+        type: 'user',
+        content: element.value,
+      },
+    ]);
+
+    element.value = '';
+  }
+
+  return (
+    <ResultContainer title="Docs Chat" description="In this demo, you can ask questions based on a document.">
```

Comment:
It would probably be good to link to the doc in question so the demo is more understandable.